### PR TITLE
[MachineVerifier] Improve checks for G_INSERT_SUBVECTOR.

### PR DESCRIPTION
**Pull request by topperc**
Cleanup messages.
Remove redundant checks that are handled in generic code.
Add overrun check.
Add checks that subvector is smaller than the vector.

### CODE REVIEW

File: llvm/test/MachineVerifier/test_g_insert_subvector.mir [hunk 1]
```diff
@@ -11,9 +11,11 @@ body:             |
     %1:_(<vscale x 2 x s32>) = G_IMPLICIT_DEF
     %2:_(<vscale x 1 x s32>) = G_IMPLICIT_DEF
 
+    ; CHECK: generic instruction must use register operands
```
Comment (michaelmaitland) [inline llvm/test/MachineVerifier/test_g_insert_subvector.mir:14]: Is this supposed to say `generic instruction must use register operands`? looks unrelated

Comment (topperc) [reply llvm/test/MachineVerifier/test_g_insert_subvector.mir:14]: anything that has a type declared in the .td automatically generates a check that the operand is a register. This handles a significant portion of the generic instructions like G_ADD, G_FADD without writing any custom code.

I was going to remove the error from G_INSERT_SUBVECTOR and G_EXTRACT_SUBVECTOR, but we need the call to `isReg()` to avoid an assert in `getReg()` later. The assert will crash the test without letting us get through all the test cases.

File: llvm/lib/CodeGen/MachineVerifier.cpp [hunk 2]
```diff
@@ -1710,41 +1710,51 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
...
     LLT DstTy = MRI->getType(MI->getOperand(0).getReg());
-    LLT Src0Ty = MRI->getType(Src0Op.getReg());
     LLT Src1Ty = MRI->getType(Src1Op.getReg());
 
     if (!DstTy.isVector()) {
       report("Destination type must be a vector", MI);
       break;
     }
 
-    if (!Src0Ty.isVector()) {
```
Comment (michaelmaitland) [inline llvm/lib/CodeGen/MachineVerifier.cpp:1721]: Is it okay to remove this check?

Comment (topperc) [reply llvm/lib/CodeGen/MachineVerifier.cpp:1721]: The relationship is already declared in the .td file and checked automatically. This check was redundant.

**Review by michaelmaitland — APPROVED** [review]
LGTM
